### ISSUE TITLE
qt5-base ignores VCPKG_BUILD_TYPE

### ISSUE THREAD

**Issue by josch** [Microsoft/vcpkg#4866]
Even with `set(VCPKG_BUILD_TYPE release)` in my tripletfile, qt5-base builds both targets: debug *and* release. Instead, it should only do the build specified by `VCPKG_BUILD_TYPE`.

This is significant because qt5-base is one of the packages that take longer to build which is critical on appveyor where builds are limited to running for only an hour. Sometimes, the build gets cancelled before qt5-base is built and then the vcpkg cache doesn't even get updated.

**Comment by josch** [comment]
In addition, to fix all qt5-* libraries, configure_qt has to be fixed as well. Here is a untested patch that somebody could use as the basis for a pull request:

```diff
diff --git a/ports/qt5-base/configure_qt.cmake b/ports/qt5-base/configure_qt.cmake
index aa3fe4af..fcae6448 100644
--- a/ports/qt5-base/configure_qt.cmake
+++ b/ports/qt5-base/configure_qt.cmake
@@ -25,6 +25,7 @@ function(configure_qt)
         set(CONFIGURE_BAT "configure")
     endif()
 
+    if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
     message(STATUS "Configuring ${TARGET_TRIPLET}-dbg")
     file(MAKE_DIRECTORY ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg)
     vcpkg_execute_required_process(
@@ -45,7 +46,9 @@ function(configure_qt)
         LOGNAME config-${TARGET_TRIPLET}-dbg
     )
     message(STATUS "Configuring ${TARGET_TRIPLET}-dbg done")
+    endif()
 
+    if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
     message(STATUS "Configuring ${TARGET_TRIPLET}-rel")
     file(MAKE_DIRECTORY ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel)
     vcpkg_execute_required_process(
@@ -65,5 +68,6 @@ function(configure_qt)
         LOGNAME config-${TARGET_TRIPLET}-rel
     )
     message(STATUS "Configuring ${TARGET_TRIPLET}-rel done")
+    endif()
 
 endfunction()
diff --git a/ports/qt5-base/install_qt.cmake b/ports/qt5-base/install_qt.cmake
index 482802f3..85fef647 100644
--- a/ports/qt5-base/install_qt.cmake
+++ b/ports/qt5-base/install_qt.cmake
@@ -28,6 +28,7 @@ function(install_qt)
     set(ENV{PATH} "${PYTHON3_EXE_PATH};$ENV{PATH}")
     set(_path "$ENV{PATH}")
 
+    if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
     message(STATUS "Package ${TARGET_TRIPLET}-dbg")
     set(ENV{PATH} "${CURRENT_INSTALLED_DIR}/debug/bin;${_path}")
     vcpkg_execute_required_process(
@@ -41,7 +42,9 @@ function(install_qt)
         LOGNAME package-${TARGET_TRIPLET}-dbg
     )
     message(STATUS "Package ${TARGET_TRIPLET}-dbg done")
+    endif()
     
+    if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
     message(STATUS "Package ${TARGET_TRIPLET}-rel")
     set(ENV{PATH} "${CURRENT_INSTALLED_DIR}/bin;${_path}")
     vcpkg_execute_required_process(
@@ -55,6 +58,7 @@ function(install_qt)
         LOGNAME package-${TARGET_TRIPLET}-rel
     )
     message(STATUS "Package ${TARGET_TRIPLET}-rel done")
+    endif()
     
     set(ENV{PATH} "${_path}")
     
diff --git a/scripts/cmake/vcpkg_configure_qmake.cmake b/scripts/cmake/vcpkg_configure_qmake.cmake
index 3aa30e70..0375f409 100644
--- a/scripts/cmake/vcpkg_configure_qmake.cmake
+++ b/scripts/cmake/vcpkg_configure_qmake.cmake
@@ -30,6 +30,7 @@ function(vcpkg_configure_qmake)
     # Cleanup build directories
     file(REMOVE_RECURSE ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg)
 
+    if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
     configure_file(${CURRENT_INSTALLED_DIR}/tools/qt5/qt_release.conf ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/qt.conf)
 
     message(STATUS "Configuring ${TARGET_TRIPLET}-rel")
@@ -40,7 +41,9 @@ function(vcpkg_configure_qmake)
         LOGNAME config-${TARGET_TRIPLET}-rel
     )
     message(STATUS "Configuring ${TARGET_TRIPLET}-rel done")
+    endif()
 
+    if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
     configure_file(${CURRENT_INSTALLED_DIR}/tools/qt5/qt_debug.conf ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/qt.conf)
 
     message(STATUS "Configuring ${TARGET_TRIPLET}-dbg")
@@ -51,5 +54,6 @@ function(vcpkg_configure_qmake)
         LOGNAME config-${TARGET_TRIPLET}-dbg
     )
     message(STATUS "Configuring ${TARGET_TRIPLET}-dbg done")
+    endif()
 
 endfunction()
\ No newline at end of file
```

**Comment by josch** [comment]
@JPMMaia how sure are you that your pull request fixes this issue? I'm on 221baafff410e4490a7cbc3f343420a8e6f3ed40 and am still experiencing this problem:

```
-- Downloading http://download.qt.io/official_releases/qt/5.12/5.12.1/submodules/qtdeclarative-everywhere-src-5.12.1.tar.xz...
-- Extracting source C:/slam6d/3rdparty/vcpkg/downloads/qtdeclarative-everywhere-src-5.12.1.tar.xz
-- Using source at C:/slam6d/3rdparty/vcpkg/buildtrees/qt5-declarative/src/5.12.1-1f64885d6f
-- Downloading https://www.python.org/ftp/python/2.7.14/python-2.7.14.msi...
CMake Error: File C:/slam6d/3rdparty/vcpkg/installed/x64-windows/tools/qt5/qt_release.conf does not exist.
CMake Error at scripts/cmake/vcpkg_configure_qmake.cmake:34 (configure_file):
  configure_file Problem configuring file
Call Stack (most recent call first):
  installed/x64-windows/share/qt5modularscripts/qt_modular_library.cmake:58 (vcpkg_configure_qmake)
  installed/x64-windows/share/qt5modularscripts/qt_modular_library.cmake:158 (qt_modular_build_library)
  ports/qt5-declarative/portfile.cmake:5 (qt_modular_library)
  scripts/ports.cmake:71 (include)
Error: Building package qt5-declarative:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: qt5-declarative:x64-windows
  Vcpkg version: 2018.11.23-nohash
```

And have `set^(VCPKG_BUILD_TYPE release^)` in my `triplets\x64-windows.cmake`. Full log:

https://ci.appveyor.com/project/josch/3dtk/builds/24177793
